arduino/WandCore/WandCore.ino

```diff
@@ -330,23 +296,32 @@ void loop()
   // Acceleration threshold reached, predict gesture
   if (aSum >= accelerationThresholdG || gSum >= gyroThreshold) {
     readAndPredictGesture(imuData, gestureConfidence);
-    int capVal = touchRead(touchPin);
     
-    // If it's a flick, turn the target light on. Twist, turn it off.
-    int lightID = 14; // CJ Room is 33, Piano Lamp is 14
-    if (gestureConfidence[GESTURE_FLICK] > 0.9) 
-    {
-      debugI("Flick! Light turning on.");
-      debugI("Touch Value is %d.", capVal);
-      myHue.setLightPower(lightID, myHue.ON);
+    if (gestureConfidence[GESTURE_FLICK] > 0.9) { // it's a flick, do flick things!
+      debugI("Flick registered. Mode is %d.", wandMode);
+      if (wandMode == POWER) myHue.setLightPower(lightID, myHue.ON); // turn on the light
+      else if (wandMode == COLOR) { // color mode, cycle through color presets
+        
+      }
+      
     }
-    else if (gestureConfidence[GESTURE_TWIST] > 0.9)
-    {
-      debugI("Twist! Light turning off.");
-      debugI("Touch Value is %d.", capVal);
-      myHue.setLightPower(lightID, myHue.OFF);
+    else if (gestureConfidence[GESTURE_TWIST] > 0.9) { // it's a twist, do twist things!    
+      debugI("Twist registered. Mode is %d", wandMode);
+      if (wandMode == POWER) myHue.setLightPower(lightID, myHue.OFF); // turn off the light
+      else if (wandMode == COLOR) { // toggle colorloop on bulb
+        if (isLightColorloopOn) {
+          myHue.setLightColorloop(lightID, myHue.OFF);
+          isLightColorloopOn = false;
+        }
+        else {
+          myHue.setLightColorloop(lightID, myHue.ON);
+          isLightColorloopOn = true;
+        }

```

Comment:
Not that I think it's that big of a deal, but this may get out of sync with reality. We don't have any reading of Hue state in here yet, not sure if we want to add it.

Reply:
yeah, we do need to add state I realized today as it's vital for incrementing/decrementing brightness. Will update this once I've implemented getting of state